qemu/tests/blockdev_inc_backup_bitmap_max_len_name.py

```diff
@@ -0,0 +1,99 @@
+import json
+import os
+import re
+
+from avocado.utils import process
+
+from provider.block_dirty_bitmap import block_dirty_bitmap_add
+from provider.block_dirty_bitmap import get_bitmap_by_name
+from provider.blockdev_live_backup_base import BlockdevLiveBackupBaseTest
+
+
+class BlockdevIncbkAddBitmapMaxLenName(BlockdevLiveBackupBaseTest):
+    """Add a bitmap with the max len name(1023 chars)"""
+
+    def __init__(self, test, params, env):
+        super(BlockdevIncbkAddBitmapMaxLenName, self).__init__(test,
+                                                               params,
+                                                               env)
+        self._max_len_name = self._make_bitmap_name()
+
+    def _make_bitmap_name(self):
+        length = self.params.get_numeric(
+            'bitmap_name_max_len') - len(self.params['prefix_name'])
+        return self.params['prefix_name'] + process.run(
+            self.params['create_bitmap_name_cmd'].format(length=length),
+            ignore_status=True, shell=True
+        ).stdout.decode().strip()
+
+    def prepare_test(self):
+        self.preprocess_data_disks()
+        self.prepare_main_vm()
+
+    def add_persistent_bitmap(self):
+        kargs = {'bitmap_name': self._max_len_name,
+                 'target_device': self._source_nodes[0],
+                 'persistent': 'on'}
+        block_dirty_bitmap_add(self.main_vm, kargs)
+
+    def check_image_bitmap_qemu_img(self):
+        data_image_obj = self.source_disk_define_by_params(
+            self.params, self._source_images[0])
+        try:
+            out = json.loads(data_image_obj.info(True, 'json'))
+            bitmap = out['format-specific']['data']['bitmaps'][0]
+        except Exception as e:
+            self.test.fail('Failed to get bitmap: %s' % str(e))
+        else:
+            if bitmap['name'] != self._max_len_name:
+                self.test.fail('Failed to get bitmap with qemu-img')
+
+    def check_image_bitmap_with_qmp_cmd(self):
+        bitmap = get_bitmap_by_name(self.main_vm, self._source_nodes[0],
+                                    self._max_len_name)
+        if bitmap is None:
+            self.test.fail('Failed to get bitmap with query-block')
+
+    def check_qemu_aborted(self):
+        """We used to hit core once, so add this check for future detection"""
+        log_file = os.path.join(self.test.resultsdir,
+                                self.params.get('debug_log_file', 'debug.log'))
+        with open(log_file, 'r') as f:
+            out = f.read().strip()
+            if re.search(self.error_msg, out, re.M):
+                self.test.fail('qemu aborted (core dumped)')
```

Comment:
Could we use the `self.main_vm.process `? like:
```
            # Make sure the process was started successfully
            if not self.process.is_alive():
                output = self.process.get_output().strip()
```
instead of reading log.

Reply:
hi Yongxue, I tried get_output and found that it failed to catch the core dump information; the msg will output with logging.info, I also tried to reset the output_func, and always get  the 'ERROR: can't pickle _thread.lock objects' error, so here I have to keep the original.

Reply:
@zhencliu 
Yes, you are right, I didn't notice that qemu hit core dump after `post_test()`

Reply:
hi @YongxueHong , as we talked, now use test.logfile to get path to the debug log, it worked as expected.